[BUG]: Milvus Test 'test_multi_instance_with_collection_lock' Failing

### Version

23.11

### Which installation method(s) does this occur on?

Docker

### Describe the bug.

Milvus test `test_multi_instance_with_collection_lock` is failing as the threadpool executor doesn't guarantee the order of execution.

```
=========================== short test summary info ============================
FAILED test_milvus_vector_db_service.py::test_multi_instance_with_collection_lock - AssertionError: assert ['Create Exec...ert Executed'] == ['Create Exec...rch Executed']
  At index 1 diff: 'Search Executed' != 'Insert Executed'
  Full diff:
  - ['Create Executed', 'Insert Executed', 'Search Executed']
  ?                      ^^^  ^             ^ - ^^
  + ['Create Executed', 'Search Executed', 'Insert Executed']
  ?                      ^ + ^^             ^^^  ^
==== 1 failed, 2491 passed, 17 skipped, 191 warnings in 1348.53s (0:22:28) =====
```

### Minimum reproducible example

_No response_

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

 [Paste the error here, it will be hidden by default]

</pre></details>


### Full env printout

<details><summary>Click here to see environment details</summary><pre>

 [Paste the results of print_env.sh here, it will be hidden by default]

</pre></details>


### Other/Misc.

_No response_

### Code of Conduct

- [X] I agree to follow Morpheus' Code of Conduct
- [X] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

--- a/morpheus/service/milvus_vector_db_service.py
+++ b/morpheus/service/milvus_vector_db_service.py
@@ -148,6 +148,11 @@ class MilvusVectorDBService(VectorDBService):
         ------
         ValueError
             If the provided schema fields configuration is empty.
+
+        Notes
+        -----
+        This function is decorated with `with_collection_lock` to ensure that the operation is synchronized by acquiring
+        and releasing a collection-specific lock, preventing concurrent access issues in the Milvus vector database.
         """
         logger.debug("Creating collection: %s, overwrite=%s, kwargs=%s", name, overwrite, kwargs)
         # Preserve original configuration.
@@ -219,6 +224,11 @@ class MilvusVectorDBService(VectorDBService):
         ------
         RuntimeError
             If the collection not exists exists.
+
+        Notes
+        -----
+        This function is decorated with `with_collection_lock` to ensure that the operation is synchronized by acquiring
+        and releasing a collection-specific lock, preventing concurrent access issues in the Milvus vector database.
         """
 
         return self._collection_insert(name, data, **kwargs)
@@ -280,6 +290,11 @@ class MilvusVectorDBService(VectorDBService):
         ------
         RuntimeError
             If the collection not exists exists.
+
+        Notes
+        -----
+        This function is decorated with `with_collection_lock` to ensure that the operation is synchronized by acquiring
+        and releasing a collection-specific lock, preventing concurrent access issues in the Milvus vector database.
         """
         if not self.has_store_object(name):
             raise RuntimeError(f"Collection {name} doesn't exist.")
@@ -362,6 +377,11 @@ class MilvusVectorDBService(VectorDBService):
         -------
         dict[str, typing.Any]
             Returns result of the updated operation stats.
+
+        Notes
+        -----
+        This function is decorated with `with_collection_lock` to ensure that the operation is synchronized by acquiring
+        and releasing a collection-specific lock, preventing concurrent access issues in the Milvus vector database.
         """
 
         if not isinstance(data, list):
@@ -389,6 +409,11 @@ class MilvusVectorDBService(VectorDBService):
         -------
         typing.Any
             Returns result of the given keys that are delete from the collection.
+
+        Notes
+        -----
+        This function is decorated with `with_collection_lock` to ensure that the operation is synchronized by acquiring
+        and releasing a collection-specific lock, preventing concurrent access issues in the Milvus vector database.
         """
 
         result = self._client.delete(collection_name=name, pks=keys, **kwargs)
@@ -413,6 +438,11 @@ class MilvusVectorDBService(VectorDBService):
         -------
         dict[str, typing.Any]
             Returns result of the given keys that are delete from the collection.
+
+        Notes
+        -----
+        This function is decorated with `with_collection_lock` to ensure that the operation is synchronized by acquiring
+        and releasing a collection-specific lock, preventing concurrent access issues in the Milvus vector database.
         """
 
         result = self._client.delete_by_expr(collection_name=name, expression=expr, **kwargs)
@@ -438,6 +468,11 @@ class MilvusVectorDBService(VectorDBService):
         -------
         list[typing.Any]
             Returns result rows of the given keys from the collection.
+
+        Notes
+        -----
+        This function is decorated with `with_collection_lock` to ensure that the operation is synchronized by acquiring
+        and releasing a collection-specific lock, preventing concurrent access issues in the Milvus vector database.
         """
 
         result = None

--- a/tests/test_milvus_vector_db_service.py
+++ b/tests/test_milvus_vector_db_service.py
@@ -14,7 +14,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import concurrent.futures
 import json
 import os
 
@@ -346,79 +345,6 @@ def test_delete(milvus_service_fixture: MilvusVectorDBService,
 
     # Clean up the collection.
     milvus_service_fixture.drop(collection_name)
-
-
-@pytest.mark.slow
-def test_single_instance_with_collection_lock(milvus_service_fixture: MilvusVectorDBService,
-                                              idx_part_collection_config_fixture: dict,
-                                              data_fixture: list[dict]):
-
-    # Create a collection.
-    collection_name = "test_insert_and_search_order_with_collection_lock"
-    milvus_service_fixture.create(collection_name, **idx_part_collection_config_fixture)
-
-    filter_query = "age == 26 or age == 27"
-    search_vec = np.random.random((1, 10))
-    execution_order = []
-
-    def insert_data():
-        result = milvus_service_fixture.insert(collection_name, data_fixture)
-        assert result['insert_count'] == len(data_fixture)
-        execution_order.append("Insert Executed")
-
-    def search_data():
-        result = milvus_service_fixture.search(collection_name, data=search_vec, filter=filter_query)
-        execution_order.append("Search Executed")
-        assert isinstance(result, list)
-
-    def count_entities():
-        milvus_service_fixture.count(collection_name)
-        execution_order.append("Count Collection Entities Executed")
-
-    with concurrent.futures.ThreadPoolExecutor(max_workers=3) as executor:
-        executor.submit(insert_data)
-        executor.submit(search_data)
-        executor.submit(count_entities)
-
-    # Assert the execution order
-    assert execution_order == ["Count Collection Entities Executed", "Insert Executed", "Search Executed"]
-
-
-@pytest.mark.slow
-def test_multi_instance_with_collection_lock(milvus_service_fixture: MilvusVectorDBService,
-                                             idx_part_collection_config_fixture: dict,
-                                             data_fixture: list[dict],
-                                             milvus_server_uri: str):
-
-    milvus_service_2 = MilvusVectorDBService(uri=milvus_server_uri)
-
-    collection_name = "test_insert_and_search_order_with_collection_lock"
-    filter_query = "age == 26 or age == 27"
-    search_vec = np.random.random((1, 10))
-
-    execution_order = []
-
-    def create_collection():
-        milvus_service_fixture.create(collection_name, **idx_part_collection_config_fixture)
-        execution_order.append("Create Executed")
-
-    def insert_data():
-        result = milvus_service_2.insert(collection_name, data_fixture)
-        assert result['insert_count'] == len(data_fixture)
-        execution_order.append("Insert Executed")
-
-    def search_data():
-        result = milvus_service_fixture.search(collection_name, data=search_vec, filter=filter_query)
-        execution_order.append("Search Executed")
-        assert isinstance(result, list)
-
-    with concurrent.futures.ThreadPoolExecutor(max_workers=3) as executor:
-        executor.submit(create_collection)
-        executor.submit(insert_data)
-        executor.submit(search_data)
-
-    # Assert the execution order
-    assert execution_order == ["Create Executed", "Insert Executed", "Search Executed"]
 
 
 def test_get_collection_lock():

